src/main/java/duke/command/MarkCommand.java

```diff
@@ -0,0 +1,35 @@
+package duke.command;
+
+import duke.Storage;
+import duke.exception.InvalidIndexException;
+import duke.task.TaskList;
+
+/**
+ * Represents a Mark Command
+ */
+public class MarkCommand extends Command {
+    private boolean isMark;
+    private int index;
+
+    /**
+     * Creates a Mark Command object
+     */
+    public MarkCommand(boolean isMark, int index) throws InvalidIndexException {
+        if (index < 0 || index > TaskList.length() - 1) {
+            throw new InvalidIndexException();
+        }
+
+        this.isMark = isMark;
+        this.index = index;
+    }
+
+    /**
+     * Marks a task in tasklist as done/not done
+     * saves tasklist to storage task file
+     */
+    @Override
+    public void execute(TaskList taskList, Storage storage) {
+        taskList.markTask(this.isMark, this.index);
```

Comment:
This isn't really a violation of coding standards, but perhaps consider omitting `this` when not necessary to follow the advanced coding guidelines?